### PR TITLE
feat(dataframe): support getting DataFrame item by index and location

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2b4rn3d 实现 DataFrame 类，支持通过 index 和 integer-location 索引数据](https://app.clickup.com/t/2b4rn3d)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -47,33 +59,52 @@ class DataFrame:
...
+        if columns is not None:
+            # TODO: missing columns processing
+            pass
+
+        if isinstance(data, dict):
+            self.columns = RowSeries(data)
+            self._data = self.columns._data
+            self.index = ColumnSeries(list(range(len(self._data[0]))))
+        else:
+            raise ValueError("DataFrame only support generate from dictionary now")
```
Comment (linjiX) [inline graviti/dataframe/frame.py:84]: `TypeError`?

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -47,33 +59,52 @@ class DataFrame:
...
         ...
 
-    def __getitem__(self, key: Union[str, Iterable[str]]) -> Union[Series, "DataFrame"]:
-        pass
+    def __getitem__(self, key: Union[str, Iterable[str]]) -> Union[ColumnSeries, "DataFrame"]:
+        value: Union[Sequence[Any], RowSeries] = self.columns[key]
+        if isinstance(value, RowSeries):
+            return self._constructer(columns=value)
+
+        return ColumnSeries(value, index=self.index._indices)
```
Comment (graczhual) [inline graviti/dataframe/frame.py:100]: name of this series

File: graviti/dataframe/frame.py (hunk 1 above)
Comment (AChenQ) [reply graviti/dataframe/frame.py:93]: Pandas will raise `ValueError`  when use the wrong data to initialize.

File: graviti/dataframe/frame.py [hunk 3]
```diff
@@ -47,36 +59,64 @@ class DataFrame:
...
+
+        self.columns = {}
+        self.column_names = []
+        if isinstance(data, dict):
+            for key, value in data.items():
+                if isinstance(value, dict):
+                    value = DataFrame(value)
+                else:
+                    value = ColumnSeries(value, name=key)
+                self.columns[key] = value
```
Comment (linjiX) [inline graviti/dataframe/frame.py:89]: `self.columns[key] = DataFrame(value) if isinstance(value, dict) else ColumnSeries(value, name=key)`

File: graviti/dataframe/frame.py [hunk 4]
```diff
@@ -47,36 +59,64 @@ class DataFrame:
...
+        if data is None:
+            data = {}
+        if schema is not None:
+            # TODO: missing schema processing
+            pass
+        if columns is not None:
+            # TODO: missing columns processing
+            pass
+
+        self.columns = {}
```
Comment (linjiX) [inline graviti/dataframe/frame.py:81]: should `columns` `column_name` and `index` be private?

File: graviti/dataframe/frame.py [hunk 5]
```diff
@@ -47,36 +59,64 @@ class DataFrame:
...
+            for key, value in data.items():
+                if isinstance(value, dict):
+                    value = DataFrame(value)
+                else:
+                    value = ColumnSeries(value, name=key)
+                self.columns[key] = value
+                self.column_names.append(key)
+            self.index = ColumnSeries(list(range(self.__len__())))
+        else:
+            raise ValueError("DataFrame only support generate from dictionary now")
```
Comment (Lee-000) [inline graviti/dataframe/frame.py:93]: `DataFrame only supports generating ...`?

File: graviti/dataframe/row/series.py [hunk 6]
```diff
@@ -54,54 +51,66 @@ def __init__(
...
-        return Series(new_data, name=self.name)
+            return self._indices_data[self._indices[key]]
+
+        indices_data = {
+            self._indices[index]: self._indices_data[self._indices[index]] for index in key
+        }
+        return self._constructer(indices_data, self.name)
+
+    @staticmethod
+    def _constructer(indices_data: Dict[str, Any], new_name: Union[str, int, None]) -> "Series":
```
Comment (linjiX) [inline graviti/dataframe/row/series.py:110]: This looks like a `classmethod`, and should `DataFrame._constructer` be a classmethod?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/row/series.py (hunk 6 above)
Comment (linjiX) [reply graviti/dataframe/row/series.py:110]: and normally we do not name a method in class in `-er` style

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 7]
```diff
@@ -47,36 +59,64 @@ class DataFrame:
...
+        new_columns = {name: self.columns[name] for name in key}
+        return self._constructer(columns=new_columns)
 
     def __setitem__(self, key: str, value: Iterable[Any]) -> None:
         pass
 
     def __repr__(self) -> str:
-        pass
+        # for test
+        return f"\n DataFrame\n columns: {self.columns} \n index: {self.index}"
```
Comment (Lee-000) [inline graviti/dataframe/frame.py:116]: Remove the test code.

**Review by Lee-000 — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**